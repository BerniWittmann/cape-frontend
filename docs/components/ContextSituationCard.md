# ContextSituationCard

A Card to view and edit a single Context Situation

## Props

<!-- @vuese:ContextSituationCard:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|contextSituation|The ContextSituation Object|`Object`|`true`|-|

<!-- @vuese:ContextSituationCard:props:end -->


## Events

<!-- @vuese:ContextSituationCard:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|deactivate|Fired when the Card should be closed|-|

<!-- @vuese:ContextSituationCard:events:end -->


