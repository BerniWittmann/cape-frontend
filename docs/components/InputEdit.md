# InputEdit

A component to edit a text but hidden behind a text with an edit button

## Props

<!-- @vuese:InputEdit:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|value|The Value of the input field|`String`|`true`|-|
|rules|The Rules array that the input field should be validated against|`Array`|`false`||
|size|Size of the controls|`'medium'` / `'small'` / `'mini'`|`false`|'normal'|
|placeholder|The Placeholder for the input field|`String`|`false`|undefined|
|propName|The name the field should be referenced with within the form|`String`|`false`|'text'|

<!-- @vuese:InputEdit:props:end -->


## Events

<!-- @vuese:InputEdit:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|change|Fired when the text changes|The new text|

<!-- @vuese:InputEdit:events:end -->


## Methods

<!-- @vuese:InputEdit:methods:start -->
|Method|Description|Parameters|
|---|---|---|
|showInput|shows the Input field|-|

<!-- @vuese:InputEdit:methods:end -->


