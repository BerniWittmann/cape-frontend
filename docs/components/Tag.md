# Tag

A Component to view a single Tag

## Props

<!-- @vuese:Tag:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|tag|The Tag object|`Object`|`true`|-|
|closable|Option whether the tag should be deleteable or not|`Boolean`|`false`|false|
|size|Size of the Tag|`'medium'` / `'small'` / `'mini'`|`false`|-|

<!-- @vuese:Tag:props:end -->


## Events

<!-- @vuese:Tag:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|close|Emitted when the close button is clicked|-|

<!-- @vuese:Tag:events:end -->


