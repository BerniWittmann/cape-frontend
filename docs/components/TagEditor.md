# TagEditor

A Component to edit a list of tags

## Props

<!-- @vuese:TagEditor:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|tags|The Array of Tag Objects|`Array`|`true`|-|

<!-- @vuese:TagEditor:props:end -->


## Events

<!-- @vuese:TagEditor:events:start -->
|Event Name|Description|Parameters|
|---|---|---|
|change|Fired when the tag array changes|The new tag array|

<!-- @vuese:TagEditor:events:end -->


