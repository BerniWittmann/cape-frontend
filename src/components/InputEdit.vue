<template>
  <div class="input-edit">
    <span v-if="!isEditing">{{ data.text }}</span>
    <span v-else>
      <el-form ref="textForm" :model="data" label-position="top" :rules="validationRules" status-icon inline :size="size"
               @submit.native.prevent="hideInput" class="form">
        <el-form-item :prop="propName">
          <el-input
                  v-model="data.text"
                  ref="inputField"
                  :size="size"
                  :placeholder="placeholder"
                  @blur="hideInput"
                  @keyup.enter.native="hideInput"
                  @submit.native.prevent="hideInput"
                  class="input"
          ></el-input>
        </el-form-item>
      </el-form>
    </span>
    <el-button
            type="text"
            icon="el-icon-edit"
            @click="showInput"
            v-if="!isEditing"
            class="black-color tag-space"
    ></el-button>
  </div>
</template>

<script>
/*
 * @vuese
 * @group Components
 *
 * A component to edit a text but hidden behind a label with an edit button
 */
export default {
  name: 'InputEdit',
  model: {
    prop: 'value',
    event: 'change'
  },

  props: {
    // The Value of the input field
    value: {
      type: String,
      required: true
    },
    // The Rules array that the input field should be validated against
    rules: {
      type: Array,
      default: () => []
    },
    // Size of the controls
    size: {
      // `'medium'` / `'small'` / `'mini'`
      type: String,
      default: 'normal'
    },
    // The Placeholder for the input field
    placeholder: {
      type: String,
      default: undefined
    },
    // The name the field should be referenced with within the form
    propName: {
      type: String,
      default: 'text'
    }
  },

  data() {
    return {
      data: {
        text: undefined
      },
      isEditing: false
    }
  },

  computed: {
    validationRules() {
      return {
        [this.propName]: this.rules
      }
    }
  },

  watch: {
    value: {
      deep: true,
      immediate: true,
      handler: function (newValue) {
        this.data.text = newValue
      }
    }
  },

  methods: {
    // @vuese
    // shows the Input field
    showInput() {
      this.isEditing = true
      this.$nextTick(() => {
        try {
          this.$refs.inputField.$refs.input.focus()
        } catch (e) {}
      })
    },

    // @vuese
    // hides the Input field
    hideInput() {
      this.$refs.textForm.validate((valid) => {
        if (valid) {
          this.isEditing = false
          // Fired when the text changes
          // @arg The new text
          this.$emit('change', this.data.text)
        }
      })
    }
  }
}

</script>

<style scoped lang="scss">
.input-edit {
  display: inline;
}
.form {
  margin-top: 10px;
  display: inline;
}
.black-color {
  color: #000000;
}

.tag-space {
  margin-left: 5px;
}
</style>
