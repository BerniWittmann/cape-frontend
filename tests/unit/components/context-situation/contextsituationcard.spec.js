import { config, mount, TransitionStub } from '@vue/test-utils'
import { i18n } from '../../setupPlugins'

import Tag from '@/components/Tag.vue'
import TagEditor from '@/components/TagEditor.vue'
import InputEdit from '@/components/InputEdit.vue'
import ContextSituationCard from '@/components/context-situation/ContextSituationCard.vue'
import ContextSituationService from '@/services/contextSituation'

describe('Components', () => {
  describe('ContextSituationCard', () => {
    let propsData
    let cmp
    let router
    let store
    config.stubs.transition = false
    console.warn = jest.fn()

    beforeEach((done) => {
      jest.useRealTimers()
      propsData = {
        contextSituation: {
          id: '142',
          name: 'My Context Situation',
          rules: 'The Rules String',
          tags: [{
            id: '43',
            name: 'Second Tag',
            color: '#FFFF00'
          }, {
            id: '44',
            name: 'Third Tag',
            color: '#0FFF00'
          }]
        }
      }
      store = {
        state: {
          tag: {
            tags: [{
              id: '42',
              name: 'First Tag',
              color: '#FF0000'
            }, {
              id: '43',
              name: 'Second Tag',
              color: '#FFFF00'
            }, {
              id: '44',
              name: 'Third Tag',
              color: '#0FFF00'
            }, {
              id: '45',
              name: 'Fourth Tag',
              color: '#00FFF0'
            }]
          },
          contextSituation: {
            activeContextSituation: {
              id: '142',
              name: 'My Context Situation',
              rules: 'The Rules String'
            }
          }
        },
        dispatch: jest.fn().mockImplementation(() => ({
          then: (arg) => arg()
        }))
      }
      router = {
        push: jest.fn(),
        back: jest.fn()
      }
      render(done)
    })

    function render(done = () => {}) {
      config.stubs.transition = TransitionStub
      cmp = mount(ContextSituationCard, {
        i18n,
        propsData,
        mocks: {
          $store: store,
          $router: router
        },
        sync: false
      })
      cmp.vm.$nextTick(done)
    }

    describe('it is not active', () => {
      beforeEach(() => {
        store.state.contextSituation.activeContextSituation = undefined
      })

      it('renders', () => {
        expect(cmp.html()).toMatchSnapshot()
      })

      it('does not contain the active class', () => {
        expect(cmp.contains('.is-active')).toBeFalsy()
      })

      it('is also inactive for other active Context Situation', () => {
        store.state.contextSituation.activeContextSituation = { id: '999999' }
        render()

        expect(cmp.vm.isActive).toBeFalsy()
        expect(cmp.contains('.is-active')).toBeFalsy()
      })

      it('displays the name', () => {
        expect(cmp.text()).toContain('My Context Situation')
      })

      it('does not display the rules', () => {
        const rules = cmp.find('contextsituationrules-stub')
        expect(rules.exists()).toBeFalsy()
      })

      it('displays the tags', () => {
        const tags = cmp.findAll(Tag)

        expect(tags.length).toEqual(2)
        expect(tags.at(0).props('tag')).toEqual(store.state.tag.tags[1])
        expect(tags.at(1).props('tag')).toEqual(store.state.tag.tags[2])
        expect(tags.at(0).props('size')).toEqual('medium')
        expect(tags.at(1).props('size')).toEqual('medium')
      })

      it('displays a shadow on hover', () => {
        expect(cmp.find('.el-card')).toBeTruthy()
        expect(cmp.find('is-hover-shadow')).toBeTruthy()
      })

      it('navigates to route when clicked', () => {
        cmp.find('.el-card').trigger('click')

        expect(router.push).toHaveBeenCalledWith({
          name: 'context_situations.single',
          params: {
            contextSituationID: '142'
          }
        })
      })
    })

    describe('it is active', () => {
      it('renders', () => {
        expect(cmp.html()).toMatchSnapshot()
      })

      it('has the active class', () => {
        expect(cmp.contains('.is-active')).toBeTruthy()
      })

      it('displays the name', () => {
        expect(cmp.text()).toContain('My Context Situation')
      })

      it('does display the rules', () => {
        const rules = cmp.find('.context-situation-rules')
        expect(rules.exists()).toBeTruthy()
      })

      it('displays the tags', () => {
        const tags = cmp.findAll(Tag)
        expect(tags.length).toEqual(2)
        expect(tags.at(0).props('tag')).toEqual(store.state.tag.tags[1])
        expect(tags.at(1).props('tag')).toEqual(store.state.tag.tags[2])
      })

      it('displays a shadow', () => {
        expect(cmp.find('.el-card')).toBeTruthy()
        expect(cmp.find('.is-always-shadow')).toBeTruthy()
      })

      it('doesnt navigate to route when clicked and active', () => {
        cmp.find('.el-card').trigger('click')
        expect(router.push).not.toHaveBeenCalled()
      })

      describe('allows to edit the name of the context situation', () => {
        it('renders the input edit component for the name', () => {
          const input = cmp.find(InputEdit)
          expect(input.exists()).toBeTruthy()
        })
        it('allows to edit the input for the title', (done) => {
          const editTitleButton = cmp.find('button.black-color')
          expect(editTitleButton.exists).toBeTruthy()
          editTitleButton.trigger('click')
          cmp.vm.$nextTick(() => {
            expect(cmp.html()).toMatchSnapshot()
            expect(cmp.find('.el-input__inner').exists).toBeTruthy()
            done()
          })
        })

        it('hides the input after editing title', (done) => {
          const editTitleButton = cmp.find('button.black-color')
          expect(editTitleButton.exists).toBeTruthy()
          editTitleButton.trigger('click')
          cmp.vm.$nextTick(() => {
            expect(cmp.html()).toMatchSnapshot()
            const input = cmp.find('.el-input__inner')
            input.setValue('My new Context Situation Name')
            input.trigger('keyup', { key: 'Enter' })
            cmp.vm.$nextTick(() => {
              expect(cmp.html()).toMatchSnapshot()
              done()
            })
          })
        })

        it('does not hide the input after editing title if its invalid', (done) => {
          const editTitleButton = cmp.find('button.black-color')
          expect(editTitleButton.exists).toBeTruthy()
          editTitleButton.trigger('click')
          cmp.vm.$nextTick(() => {
            expect(cmp.html()).toMatchSnapshot()
            const input = cmp.find('.el-input__inner')
            input.setValue('')
            input.trigger('keyup', { key: 'Enter' })
            cmp.vm.$nextTick(() => {
              expect(cmp.html()).toMatchSnapshot()
              done()
            })
          })
        })

        it('checks the input to not be empty', (done) => {
          const editTitleButton = cmp.find('button.black-color')
          editTitleButton.trigger('click')
          cmp.vm.$nextTick(() => {
            expect(cmp.html()).toMatchSnapshot()
            const input = cmp.find('.el-input__inner')
            input.setValue('')
            input.trigger('keypress', { key: 'Enter' })
            expect(cmp.html()).toMatchSnapshot()
            expect(cmp.find('.el-input__inner').exists).toBeTruthy()
            done()
          })
        })
      })

      it('renders the tag editor', () => {
        expect(cmp.contains(TagEditor)).toBeTruthy()
      })

      it('allows to reset the context situation', () => {
        expect(cmp.find('.el-button--danger.is-plain').exists()).toBeTruthy()
        cmp.vm.contextSituationData.name = 'Different'
        cmp.vm.resetCS()
        expect(cmp.vm.contextSituationData.name).toBe(propsData.contextSituation.name)
      })

      it('allows to submit the context situation', () => {
        expect(cmp.find('.el-button--success').exists()).toBeTruthy()
        ContextSituationService.update = jest.fn().mockImplementation(() => ({
          then: (arg) => arg()
        }))
        cmp.vm.saveCS()
        expect(ContextSituationService.update).toHaveBeenCalled()
      })

      it('emits a deactivate event after submitting the context situation', () => {
        expect(cmp.find('.el-button--success').exists()).toBeTruthy()
        ContextSituationService.update = jest.fn().mockImplementation(() => ({
          then: (arg) => arg(true)
        }))
        cmp.vm.saveCS()
        expect(cmp.emitted().deactivate).toBeTruthy()
      })

      describe('allows to delete the context situation', () => {
        describe('can be deleted', () => {
          beforeEach(() => {
            cmp.vm.$confirm = jest.fn().mockImplementation(() => ({
              then: (arg) => {
                return {
                  catch: () => {
                  }
                }
              }
            }))
            ContextSituationService.remove = jest.fn().mockImplementation(() => ({
              then: (arg) => arg()
            }))
            cmp.vm.$message = jest.fn()
          })
          it('deletes the active Context Situation', () => {
            ContextSituationService.getAll = jest.fn().mockImplementation(() => {
            })
            cmp.vm.$confirm = jest.fn().mockImplementation(() => ({
              then: (arg) => {
                arg()
                return {
                  catch: () => {
                  }
                }
              }
            }))
            cmp.findAll('.el-button--danger').at(1).trigger('click')
            expect(ContextSituationService.remove).toHaveBeenCalledWith(store.state.contextSituation.activeContextSituation)
            expect(router.back).toHaveBeenCalled()
            expect(ContextSituationService.getAll).toHaveBeenCalled()
          })
          it('show as confirmation dialog', () => {
            expect(cmp.html()).toMatchSnapshot()
            const button = cmp.findAll('.el-button--danger').at(1)
            expect(button.exists()).toBeTruthy()
            button.trigger('click')
            expect(cmp.vm.$confirm).toHaveBeenCalledWith('context_situation.delete.message', 'context_situation.delete.warning',
              {
                'cancelButtonText': 'context_situation.delete.cancel',
                'confirmButtonText': 'context_situation.delete.ok',
                'type': 'warning',
                'cancelButtonClass': 'is-plain el-button--info',
                'confirmButtonClass': 'el-button--danger'
              })
          })
          it('does not delete the process if canceled', () => {
            cmp.vm.$confirm = jest.fn().mockImplementation(() => ({
              then: () => {
                return {
                  catch: (arg) => {
                    arg()
                  }
                }
              }
            }))
            cmp.findAll('.el-button--danger').at(1).trigger('click')
            expect(cmp.vm.$message).toHaveBeenCalledWith({ 'message': 'context_situation.delete.cancellation', 'type': 'info' })
            expect(ContextSituationService.remove).not.toHaveBeenCalled()
          })
        })
      })
    })
  })
})
