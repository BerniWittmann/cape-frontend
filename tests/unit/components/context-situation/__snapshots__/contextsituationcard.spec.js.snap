// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContextSituationCard it is active allows to delete the context situation can be deleted show as confirmation dialog 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span class="title">My Context Situation</span> <button type="button" class="el-button black-color tag-space el-button--text">
              <!----><i class="el-icon-edit"></i>
              <!----></button></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active allows to edit the name of the context situation allows to edit the input for the title 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span><div class="el-form-item el-form-item--feedback is-required"><!----><div class="el-form-item__content"><div class="input el-input el-input--mini"><!----><input type="text" autocomplete="off" placeholder="context_situation.edit.name" class="el-input__inner"><!----><!----><!----></div><!----></div></div></span>
            <!----></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active allows to edit the name of the context situation checks the input to not be empty 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span><div class="el-form-item el-form-item--feedback is-required"><!----><div class="el-form-item__content"><div class="input el-input el-input--mini"><!----><input type="text" autocomplete="off" placeholder="context_situation.edit.name" class="el-input__inner"><!----><!----><!----></div><!----></div></div></span>
            <!----></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active allows to edit the name of the context situation checks the input to not be empty 2`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span><div class="el-form-item el-form-item--feedback is-required"><!----><div class="el-form-item__content"><div class="input el-input el-input--mini"><!----><input type="text" autocomplete="off" placeholder="context_situation.edit.name" class="el-input__inner"><!----><!----><!----></div><!----></div></div></span>
            <!----></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active allows to edit the name of the context situation does not hide the input after editing title if its invalid 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span><div class="el-form-item el-form-item--feedback is-required"><!----><div class="el-form-item__content"><div class="input el-input el-input--mini"><!----><input type="text" autocomplete="off" placeholder="context_situation.edit.name" class="el-input__inner"><!----><!----><!----></div><!----></div></div></span>
            <!----></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active allows to edit the name of the context situation hides the input after editing title 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span><div class="el-form-item el-form-item--feedback is-required"><!----><div class="el-form-item__content"><div class="input el-input el-input--mini"><!----><input type="text" autocomplete="off" placeholder="context_situation.edit.name" class="el-input__inner"><!----><!----><!----></div><!----></div></div></span>
            <!----></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is active renders 1`] = `
<div class="el-card is-always-shadow context-situation-card is-active">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span class="title">My Context Situation</span> <button type="button" class="el-button black-color tag-space el-button--text">
              <!----><i class="el-icon-edit"></i>
              <!----></button></span> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-close"></i>
              <!----></button>
            <!---->
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags">
      <div><span class="tag el-tag el-tag--info" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag<i class="el-tag__close el-icon-close"></i></span><span class="tag el-tag el-tag--info" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag<i class="el-tag__close el-icon-close"></i></span>
        <!----> <button type="button" class="el-button tag-space el-button--default el-button--small">
          <!----><i class="el-icon-plus"></i><span>process.edit.add_tag
  </span></button></div>
    </div>
    <form class="el-form context-situation-rules top-space el-form--label-top el-form--inline">
      <div class="el-form-item use-space el-form-item--feedback">
        <!---->
        <div class="el-form-item__content"><span><span>The Rules String</span> <button type="button" class="el-button black-color tag-space el-button--text">
            <!----><i class="el-icon-edit"></i>
            <!----></button></span>
          <!---->
        </div>
      </div>
    </form>
    <div class="top-space"><button type="button" class="el-button el-button--danger el-button--medium is-plain">
        <!---->
        <!----><span>
      context_situation.edit.reset
    </span></button> <button type="button" class="el-button el-button--danger el-button--medium">
        <!---->
        <!----><span>
      context_situation.edit.delete
    </span></button> <button type="button" class="el-button el-button--success el-button--medium" style="float: right;">
        <!---->
        <!----><span>context_situation.edit.save
    </span></button></div>
  </div>
</div>
`;

exports[`Components ContextSituationCard it is not active renders 1`] = `
<div class="el-card is-hover-shadow context-situation-card">
  <div class="el-card__header">
    <div class="clearfix">
      <form class="el-form el-form--label-top el-form--inline">
        <div class="el-form-item use-space el-form-item--feedback">
          <!---->
          <div class="el-form-item__content"><span><span class="title">My Context Situation</span></span>
            <!----> <button type="button" class="el-button el-button--text" style="float: right;">
              <!----><i class="el-icon-arrow-right"></i>
              <!----></button>
            <!---->
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="el-card__body">
    <div class="tags"><span class="tag el-tag el-tag--info el-tag--medium" style="background-color: rgba(255, 255, 0, 0.1); color: rgb(255, 255, 0);" name="el-zoom-in-center">Second Tag</span><span class="tag el-tag el-tag--info el-tag--medium" style="background-color: rgba(15, 255, 0, 0.1); color: rgb(15, 255, 0);" name="el-zoom-in-center">Third Tag</span></div>
    <!---->
    <!---->
  </div>
</div>
`;
